include/rest-request.php

```diff
@@ -130,4 +131,24 @@ public function set_language( $result, $server, $request ) {
 
 		return $result;
 	}
+
+	/**
+	 * Sets the current locale during a REST request if `lang` parameter is set.
+	 *
+	 * @since 3.7.4
+	 *
+	 * @param string $locale The current locale.
+	 * @return string The current locale.
+	 */
+	public function set_locale( $locale ) {
+		if ( empty( $_REQUEST['lang'] ) || ! is_string( $_REQUEST['lang'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			return $locale;
+		}
+
+		if ( empty( $this->curlang ) || $this->curlang->slug !== $_REQUEST['lang'] ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			return $locale;
+		}
+
+		return $this->curlang->locale;
```

Comment:
Should we check for this ?
```suggestion
		return $this->curlang->locale ?? $locale;

```